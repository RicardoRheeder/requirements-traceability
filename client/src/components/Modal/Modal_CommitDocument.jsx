import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { setModalObject } from '../../redux/stores/common/actions'
import { useAuth0 } from '@auth0/auth0-react'
import {
  fetchUserDocsAsync,
  commitTreeAsync,
} from '../../redux/stores/document/actions'

export default function Modal_CommitDocument() {
  const { user } = useAuth0()

  const storeTreeData = useSelector((state) => state.common.treeData, [])
  const selectedDocObject = useSelector((state) => state.document.current_doc)

  const dispatch = useDispatch()
  const userInfo = useSelector((state) => state.user.info)

  const [doc, setDoc] = useState({
    title: '',
  })

  const handleSubmit = (e) => {
    e.preventDefaults
  }

  const commitDocumentToDB = () => {
    let docObject = { tree: JSON.stringify(storeTreeData) }
    let docID = selectedDocObject
    // committing the tree to the db (adding it to the versions array)
    dispatch(commitTreeAsync(docObject, docID, doc.title))
    dispatch(fetchUserDocsAsync(user))
  }

  const handleChange = (e) => {
    const { value } = e.target
    setDoc({ ...doc, title: value })
  }

  return (
    <div className="modal-root modal-root-child">
      <div className="modal-contents-container">
        <h1 className="modal-contents-title">Commit document</h1>
        <form onSubmit={handleSubmit}>
          {/* <h2>Please enter a commit message.</h2>
          <input className="modal-input" onChange={handleChange} /> */}
          <h2>Please enter a version number.</h2>
          <input className="modal-input" onChange={handleChange} />

          <div className="button-container">
            <button
              className="orange-button modal-button"
              onClick={commitDocumentToDB}
            >
              Submit
            </button>
            <button
              className="orange-button modal-button"
              onClick={() =>
                dispatch(setModalObject({ visible: false, mode: 0 }))
              }
            >
              Close
            </button>
          </div>
        </form>
      </div>
    </div>
  )
}
