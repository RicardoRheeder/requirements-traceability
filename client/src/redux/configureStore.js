import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
