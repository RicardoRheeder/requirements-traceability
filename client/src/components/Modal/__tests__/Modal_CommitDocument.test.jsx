import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import Modal_CommitDocument from "../Modal_CommitDocument"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("commit document modal component", () => {
    let store
    beforeEach(() => {
        store = mockStore({
            document: {
                documents: [{title: 'testdoc', versions: [], tree: null}],
                versions: [],
                current_doc: {id: "333",
                    collaborators: [
                    { _id: '111', username: 'testUser1' },
                    { _id: '222', username: 'testUser2' },
                  ]}
            },
            common: {
                selectedDocumentPanelObject: {title: 'testdoc'},
                treeData: {test:"test"}
            },
            user: {
                info: "testinfo",
                notifications: []
            }
            
        });
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><Modal_CommitDocument/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><Modal_CommitDocument/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});