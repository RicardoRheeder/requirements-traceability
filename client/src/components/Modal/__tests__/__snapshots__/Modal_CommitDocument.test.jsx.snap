// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commit document modal component renders 1`] = `ShallowWrapper {}`;

exports[`commit document modal component renders with children 1`] = `ReactWrapper {}`;
